### PR TITLE
Add support for filtering and pagination on job output

### PR DESCRIPTION
**Pull request by mabashian**
##### SUMMARY
link https://github.com/ansible/awx/issues/6612
link https://github.com/ansible/awx/issues/5906

This PR adds the ability to filter job events and also includes logic to handle fetching filtered job events across different pages.

Note that the `verbosity` dropdown included in #5906 is not included in this work.  I don't think that's possible without api changes.

As part of this work, I converted JobOutput.jsx from a class based component to a functional component.  I've tried my best to make sure that all existing functionality has remained the same by comparing the experience of this branch to devel.

Like the old UI, the output filter is disabled while the job is running.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - UI

### CODE REVIEW

File: awx/ui_next/src/screens/Job/JobOutput/shared/jobOutputUtils.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+export default function getRowRangePageSize(startIndex, stopIndex) {
+  let page;
+  let pageSize;
+
+  if (startIndex === stopIndex) {
+    page = startIndex;
```
Comment (jakemcdermott) [inline awx/ui_next/src/screens/Job/JobOutput/shared/jobOutputUtils.js]: I think there's an off-by-one here.
```suggestion
    page = startIndex + 1;
```

A reproducer for the bug is when `startIndex = 50` and `stopIndex = 50`:

Screenshot from 2021-02-02 15-20-50

**Comment by jakemcdermott** [comment]
@mabashian 
Based on what I can see from my testing, this PR resolves the issue with job results not updating until finished, (e.g: https://github.com/ansible/awx/issues/9112).

It would be good to make a note of it in the [changelog](https://github.com/ansible/awx/issues/9112)

File: awx/ui_next/src/screens/Job/JobOutput/JobOutput.jsx [hunk 2]
```diff
@@ -447,174 +468,300 @@ class JobOutput extends Component {
...
+      onReplaceSearch={handleReplaceSearch}
+      onShowAdvancedSearch={() => {}}
+      onRemove={handleRemoveSearchTerm}
+      isDisabled={isJobRunning(job.status)}
+    />
+  );
+
+  if (hasContentLoading) {
+    return <ContentLoading />;
+  }
```
Comment (marshmalien) [inline awx/ui_next/src/screens/Job/JobOutput/JobOutput.jsx]: To reduce "flashing" during search, we could move `<ContentLoading/>` further down into the Job Output. 

```
<AutoSizer nonce={window.NONCE_ID} onResize={handleResize}>
  {({ width, height }) => {
    return (
      <>
      {hasContentLoading ? (
        <div style={{ width }}>
          <ContentLoading />
        </div>
      ) : (
        <List
          ref={ref => {
            registerChild(ref);
            listRef.current = ref;
          }}
 ...
```
output

**Comment by mabashian** [comment]
@jakemcdermott @marshmalien OK, I think I've got this one rebased and ready to go with y'alls suggested changes.

**Review by jakemcdermott — APPROVED** [review]

**Comment by jakemcdermott** [comment]
@mabashian lgtm :+1: note: There's another merge conflict from a recent pr

**Review by marshmalien — APPROVED** [review]
This is awesome. Thanks for refactoring the JobOutput into a functional component - that work will go a long way.

**Comment by unlikelyzero** [comment]
Note: this is our highest priority this week.

@mabashian can you rebase?

**Comment by one-t** [comment]
Hey, folks- found this issue while I was poking around:

If you provide the wrong type of input for a filter you will have to navigate out and back to the job details page to start over
  * Steps:
    * Open a project update
    * do an advanced search, with `project_update` as the key
    * provide a non-integer value (`a`)
    * image

**Comment by dimovstanimir** [comment]
Guys aren't you going to move forward with this?

**Comment by dimovstanimir** [comment]
@unlikelyzero do you think that this feature could be ready and included in newest AWX release?

**Comment by unlikelyzero** [comment]
@mabashian "while you're in there" can you update the locators identified in https://github.com/ansible/awx/issues/9495

**Comment by mabashian** [comment]
In reference to #9495, I've added a commit to add ouiaId's to the following things:

Relaunch (Dropdown version which allows users to select all or just failed hosts):

Screen Shot 2021-03-18 at 2 19 59 PM

Relaunch (Regular button version):

Screen Shot 2021-03-18 at 2 21 21 PM

Download:

Screen Shot 2021-03-18 at 2 22 01 PM

Delete:

Screen Shot 2021-03-18 at 2 22 56 PM

Cancel:

Screen Shot 2021-03-18 at 2 23 36 PM

**Comment by unlikelyzero** [comment]
## Findings

- Host Details Standard Out and Standard Error always appear empty (User error?)
Screen Shot 2021-03-19 at 8 59 51 AM

- ~I don't appear to be able to filter on 'Event (event) Host OK'. Host Unreachable works~

- ~Cancel Job button appears after JT completes (regression)~

**Comment by unlikelyzero** [comment]
@mabashian any chance you could add data-ouia-ids on the page controls?

**Pull request merged**